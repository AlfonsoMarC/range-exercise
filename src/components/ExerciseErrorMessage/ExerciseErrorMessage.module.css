.container {
  display: flex;
  flex-direction: column;
  flex: 1;
  align-items: center;
  justify-content: center;
  font-size: var(--font-size-s);
  color: var(--color-semantic-warning);
}
