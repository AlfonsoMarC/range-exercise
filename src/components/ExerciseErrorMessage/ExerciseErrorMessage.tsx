import React from "react";
import styles from "./ExerciseErrorMessage.module.css";

export interface ExerciseErrorMessageProps {
  error?: string;
  discreteMode?: boolean;
}

const ExerciseErrorMessage: React.FC<ExerciseErrorMessageProps> = ({ error, discreteMode }) => {
  const dataName = discreteMode ? "values" : "limits";
  const errorMessage = `${
    error ?? `Invalid range ${dataName}`
  }. Default ${dataName} are being used.`;
  return (
    <div className={`${styles.container} exercise-error-message__container`}>{errorMessage}</div>
  );
};

export default ExerciseErrorMessage;
