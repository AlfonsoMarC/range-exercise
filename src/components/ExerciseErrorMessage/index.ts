import ExerciseErrorMessage, { ExerciseErrorMessageProps } from "./ExerciseErrorMessage";
export type { ExerciseErrorMessageProps };
export default ExerciseErrorMessage;
