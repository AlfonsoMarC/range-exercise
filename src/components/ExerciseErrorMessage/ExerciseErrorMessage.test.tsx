import React from "react";
import { render } from "@testing-library/react";
import ExerciseErrorMessage from "./ExerciseErrorMessage";

describe("ExerciseErrorMessage", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(<ExerciseErrorMessage error="Test Error Message" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
