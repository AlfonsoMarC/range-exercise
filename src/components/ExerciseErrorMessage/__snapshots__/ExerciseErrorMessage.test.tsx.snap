// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseErrorMessage should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container exercise-error-message__container"
  >
    Test Error Message. Default limits are being used.
  </div>
</DocumentFragment>
`;
